exporters/prometheus/exporter.go

```diff
@@ -316,9 +318,12 @@ var unitSuffixes = map[string]string{
 	"ms": "_milliseconds",
 }
 
-// getName returns the sanitized name, including unit suffix.
+// getName returns the sanitized name, prefixed with the namespace and suffixed with unit.
 func (c *collector) getName(m metricdata.Metrics) string {
 	name := sanitizeName(m.Name)
+	if c.namespace != "" {
+		name = c.namespace + name
+	}
```

Comment:
Should we do this before the name is sanitized?

Reply:
That is a good question but I've opted to sanitize the namespace once when in the `WithNamespace()` method, so `c.namespace` here is already sanitized. This avoid the need to sanitize the namespace multiple times.